kotlinx-coroutines-core/jvm/src/internal/StackTraceRecovery.kt

```diff
@@ -165,7 +165,7 @@ internal actual fun <E : Throwable> unwrap(exception: E): E {
 }
 
 private fun <E : Throwable> recoveryDisabled(exception: E) =
-    !RECOVER_STACKTRACES || !DEBUG || exception is CancellationException || exception is NonRecoverableThrowable
+    !RECOVER_STACKTRACES || !DEBUG || exception is NonRecoverableThrowable
```

Comment:
This change should be either reverted or properly integrated with JCE and tests for #950

Reply:
This change actually fixes #950. The exception becomes traceable. I'll add a test.